### ISSUE TITLE
How can I prevent the channels from skipping when the channel is frozen?

### ISSUE THREAD

**Issue by gerek33** [zaps166/QMPlay2#673]
Every time the channel starts buffering, QMPlay skips to the next channel, which is really bothersome. I tried using the "stop after playing" option, but it didn't work; it just stops the channel when it freezes. So, how can I fix this problem?

**Comment by zaps166** [comment]
Don't understand, what is the "channel"?

**Comment by gerek33** [comment]
I mean while I am watching the ip-tv.

**Comment by mtktsl** [comment]
The problem occurs when you stream media from web from a provider that is providing media in m3u format. When streaming media from m3u files, it skips to next channel if it stutters when playing media. That is the problem gerek33 is trying to explain. @zaps166

**Comment by gerek33** [comment]
I totally agree!

**Comment by zaps166** [comment]
I need to reproduce it locally, otherwise can't fix anything...

**Comment by zaps166** [comment]
I sit possible to simulate it somehow?

**Comment by gerek33** [comment]
> I sit possible to simulate it somehow?

Sure I will send a video today.

**Comment by Tinoo46** [comment]
You should check if the "Play next entry after playback error" is checked or not in the Playback Settings (Options->Settings->Playback Settings

**Comment by blzos** [comment]
> Every time the channel starts buffering, QMPlay skips to the next channel, which is really bothersome. I tried using the "stop after playing" option, but it didn't work; it just stops the channel when it freezes. So, how can I fix this problem?


To disable moving to the next file if a problem occurs during playback, you need to set two options:

Option -> Playback settings -> Play next entry after playback error [Disable]
Player -> Repeat -> Stop playback after every file [Alt+8] (Choose this option)

In this case, the problematic stream will be closed, the player will stop playing completely

Unfortunately, the option "Stop playback after every file" to reset after closing the player, on "No repeating" (BUG?)

**Comment by zaps166** [comment]
> Unfortunately, the option "Stop playback after every file" to reset after closing the player, on "No repeating" (BUG?)

`General settings`->`Remember repeat mode`.

I know, settings are hard to find (I had to look at code to find it :rofl:)

> Play next entry after playback error 

This option is disabled by default, so it shouldn't be a problem.

> stop after playing

I reproduced the issue by disconnecting network and seeking. This option works to me (stops playback), but resets after single usage (I don't remember the logic). I can make this option persistent.

Additionally I had to know the error - if it's EOF, it's obvious that the player will stop. Normally buffering works also when I lose network connection for a while. Maybe I could set some parameters, but I have to reproduce the issue.
